source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -266,19 +214,28 @@ void *ns_dyn_mem_alloc(int16_t alloc_size)
 
 
         if (mem_stat_info_ptr) {
-            alloc_size = (alloc_sector * sizeof(int));
+            // If alloc fail needs to collect size of failed alloc,
+            // this code needs to be revisited
+            alloc_size = ((alloc_sector + 2) * sizeof(int));
```

Comment:
Interesting. How do we want failures logged? At present each failure adds "rounded up size". Adding 0 doesn't work - it doesn't affect the stat!

You could just update by 1, but I guess I'd try to roughly retain previous behaviour by returning the /original/ alloc request size.

So move this into the if (ret_val), and let the failure return the /original/ alloc_size passed in.
Better still, don't actually modify the original alloc_size (it's confusing) - just put the expression in the OK stat update.

Maybe worth marking the alloc_size parameter "const" to discourage internal tinkering.